langchain4j-azure-cosmos-no-sql/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/no/sql/AzureCosmosDbNoSqlEmbeddingStore.java

```diff
@@ -0,0 +1,173 @@
+
+    @Override
+    public String add(Embedding embedding) {
+        String id = randomUUID();
+        add(id, embedding);
+        return id;
+    }
+
+    @Override
+    public void add(String id, Embedding embedding) {
+        addInternal(id, embedding, null);
+    }
+
+    @Override
+    public String add(Embedding embedding, TextSegment textSegment) {
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(Collectors.toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(Collectors.toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        String referenceEmbeddingString = referenceEmbedding.vectorAsList().stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(","));
+
+        String query = String.format("SELECT TOP %d c.id, c.%s, c.text, c.metadata, VectorDistance(c.%s,[%s]) AS score FROM c ORDER By " +
+                "VectorDistance(c.%s,[%s])", maxResults, embeddingKey, embeddingKey, referenceEmbeddingString, embeddingKey, referenceEmbeddingString);
+
+        CosmosPagedIterable<AzureCosmosDbNoSqlMatchedDocument> results = this.container.queryItems(query,
+                new CosmosQueryRequestOptions(), AzureCosmosDbNoSqlMatchedDocument.class);
+
+        if (!results.stream().findAny().isPresent()) {
+            return new ArrayList<>();
+        }
+        return results.stream()
+                .map(MappingUtils::toEmbeddingMatch)
+                .collect(Collectors.toList());
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("do not add empty embeddings to Azure CosmosDB NoSQL");
+            return;
+        }
+
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(), "embeddings size is not equal to embedded size");
+
+        PartitionKeyDefinition partitionKey = this.containerProperties.getPartitionKeyDefinition();
```

Comment:
`partitionKey` is not used, is it intended?

Reply:
Removed